splunk-otel-android/src/main/java/com/splunk/rum/ZipkinToDiskSender.java

```diff
@@ -46,6 +42,10 @@ public int messageSizeInBytes(List<byte[]> encodedSpans) {
 
     @Override
     public Call<Void> sendSpans(List<byte[]> encodedSpans) {
+        if (!storageLimiter.ensureFreeSpace()) {
```

Comment:
Nit: you could roughly estimate the amount of space needed by summing the sizes of `encodedSpans`

Reply:
Sure, you could. I thought it would be simpler to just keep deleting until we're back under the limit, and I think it's ok if a single write takes us back over the limit.  This means that the limit is a "soft limit" that we try for but are not strict about. I hope this is ok because we're talking about a whopping 25MB. :)